tools/topology/sof-icl-da7219.m4

```diff
@@ -0,0 +1,184 @@
+#
+# Topology for IceLake with Dialog7219.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Icelake DSP configuration
+include(`platform/intel/icl.m4')
+include(`platform/intel/dmic.m4')
+
+define(`SPK_INDEX', `1')
+define(`SPK_NAME', `SSP1-Codec')
+
+#
+# Define the pipelines
+#
+# PCM0  ----> volume (pipe 4)   -----> SSP1 (speaker - maxim98373, BE link 0)
+# PCM1  <---> volume (pipe 1,2) <----> SSP0 (headset - da7219, BE link 1)
+# PCM99 <---- DMIC0 (pipe 99)   <----- dmic (DMIC01 - BE link 2)
```

Comment:
just curious - why 99?

Reply:
We used to use PCM99 on glk to designate the dmic01 pcm, do not know the background. Will change the PCM #  to a sequential number instead.

Reply:
@lyakh @yongzhi1 this can be anything. Back in the GLK days, we were playing with the PCM device ID's to see if we could resuse the PCM ID from a playback-only DAI for the DMIC and obviously it didnt work. So just to make it explicit, we used 99. So just the silliest of reasons that it started out like that and it ended up confusing everyone.

So, in short, the only requirement is that the PCM ID needs to be unique

Reply:
to be more precise, the device number needs to be unique if it's playback or capture only. You can reuse device numbers if they are playback AND capture.

Reply:
Thanks for the clarification, updated https://github.com/yongzhi1/sof/commit/b36673c217a2468046470affa5a0a4c1ea99bb33

Do not know how to regenerate PR suggested by @lyakh